libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -0,0 +1,184 @@
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+"""Test library for query deployment related operations."""
+
+from http import HTTPStatus
+import yaml
+
+from config import PROV_TEST_CFG
+from commons import configmanager
+from commons.constants import LOCAL_SOLUTION_PATH
+from libs.csm.csm_interface import csm_api_factory
```

Comment:
Libs should not have csm_api_factory. It is meant to be used outside libs

Reply:
removed